pydicom/tag.py

```diff
@@ -25,43 +26,40 @@ def tag_in_exception(tag):
     """
     try:
         yield
-    except Exception as ex:
+    except Exception as exc:
         stack_trace = traceback.format_exc()
-        msg = 'With tag {0} got exception: {1}\n{2}'.format(
-            tag,
-            str(ex),
-            stack_trace)
-        raise type(ex)(msg)
+        msg = f"With tag {tag} got exception: {str(exc)}\n{stack_trace}"
+        raise type(exc)(msg) from exc
 
 
-def Tag(arg, arg2=None):
+def Tag(
+    arg: Union[int, str, Tuple[int]],
+    arg2: Optional[int] = None
+) -> "BaseTag":
     """Create a :class:`BaseTag`.
 
     General function for creating a :class:`BaseTag` in any of the standard
     forms:
 
     * ``Tag(0x00100015)``
-    * ``Tag('0x00100015')``
     * ``Tag((0x10, 0x50))``
-    * ``Tag(('0x10', '0x50'))``
     * ``Tag(0x0010, 0x0015)``
-    * ``Tag(0x10, 0x15)``
-    * ``Tag(2341, 0x10)``
-    * ``Tag('0xFE', '0x0010')``
     * ``Tag("PatientName")``
```

Comment:
I simplified these to have four "official" ways to create a BaseTag. Obviously the other forms will still work.